### PR TITLE
Fix/searxng

### PR DESCRIPTION
**Pull request by nirabo**
# Pull Request Checklist

### Fixed

The SearXNG search function has been updated to correctly construct the query string using the params argument of the requests.get() function. This ensures that the query parameters are properly encoded and included in the URL.

### CODE REVIEW

**Comment by tjbck** [comment]
LGTM, Thanks!

**Comment by tjbck** [comment]
@nirabo Error reported on Discord:

Web search with my local SearXNG with - json enabled in the settings .yaml file has been working on v0.2.

Is anyone else having trouble with Web Search failing after updating (Docker) to v0.2.3?
I get: Something went wrong :/ [Errno -3] Temporary failure in name resolution

I double checked my settings and they are the same as before with URL set to:
https://myaddressofsearxng.com/search?q=<query>

With the SearXNG instance allowed to output in - json and not blocking bots and crawlers requests.

Browser says: Failed to load resource: the server responded with a status of 400 (Bad Request)

image

File: backend/apps/rag/search/searxng.py [hunk 1]
```diff
@@ -1,44 +1,79 @@
...
+    # Default values for optional parameters are provided as empty strings or None when not specified.
+    language = kwargs.get('language', 'en-US')
+    time_range = kwargs.get('time_range', '')
+    categories = ''.join(kwargs.get('categories', []))
+
+    params = {
+        "q": query,
+        "format": "json",
+        "pageno": 1,
+        "results_per_page": count,
```
Comment (tjbck) [inline backend/apps/rag/search/searxng.py:45]: `results_per_page` doesn't seem to be a real param, which caused the error. 

https://github.com/searx/searx/issues/1276

File: backend/apps/rag/search/searxng.py [hunk 2]
```diff
@@ -1,44 +1,79 @@
...
+    response.raise_for_status()  # Raise an exception for HTTP errors.
+
+    json_response = response.json()
     results = json_response.get("results", [])
     sorted_results = sorted(results, key=lambda x: x.get("score", 0), reverse=True)
     return [
         SearchResult(
             link=result["url"], title=result.get("title"), snippet=result.get("content")
         )
-        for result in sorted_results[:count]
```
Comment (tjbck) [inline backend/apps/rag/search/searxng.py:43]: This part should've stayed. Next time, please test your PR thoroughly before creating one, Thanks!

Comment (nirabo) [reply backend/apps/rag/search/searxng.py:43]: Will do, thanks for straightening this out that promptly.

**Pull request merged**